### ISSUE TITLE
Support for enum type

### ISSUE THREAD

**Issue by Halky** [Rhetos/Rhetos#147]
**Is your feature request related to a problem? Please describe.**
It would be nice to have elegant way to encapsulate key-value pars inside structure, like enums (and maybe even remove the need to ask database for values of keys).

**Describe the solution you'd like**
```
Entity Status
{
    ShortString Name;
    ShortString Description;
    Enum 'CarStatus' { status => new Enum { Key = status.ID, Value = status.Name } };
}
```
At the end of the deployment process Rhetos would generate enum from status by pairing all the current IDs with theirs Name.
The usage would look like
```
var carToRent = repository.Schema1.Car.Query(c => c.ID == carId).First();
carToRent.StatusID = CarStatus.Rented;
repository.Schema1.Car.Update(carToRent);
```

**Describe alternatives you've considered**
Following the example above, adding flag to each status in Status table. However, sometimes it feels like writing unnecessary code just to fetch ID.
`carToRent.StatusID = repository.Schema1.Status(s => s.IsRented).Select(s => s.ID).First();`

**Comment by bantolov** [comment]
It's an interesting idea.
A similar feature is currently in development - HardCoded entity (#135), maybe this two concepts could be combined in some way.

**Comment by ibarban** [comment]
I think that the idea is the same as in the mentioned pull request.
You can see the usage in the last test in this file [](https://github.com/Rhetos/Rhetos/pull/135/files#diff-7f263282547cf24bd396caa60fb1034d).

**Comment by bantolov** [comment]
**Hardcoded** entity is implemented in latest release Rhetos v2.12, so I'm closing this issue as resolved.
The documentation is available [here](https://github.com/Rhetos/Rhetos/wiki/simple-read-only-entities-and-codetables).

Usage in C# code is same as the snippet in the original question above (`carToRent.StatusID = CarStatus.Rented`).
